### PR TITLE
Add the privileged extension

### PR DESCRIPTION
**Pull request by galou**
Signed-off-by: Gaël Écorchard <gael.ecorchard@cvut.cz>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,30 +31,31 @@
 
 kwargs = {
     'name': 'rocker',
-    'version': '0.2.3',
+    'version': '0.3.0',
```
Comment (tfoote) [inline setup.py:34]: please don't change the version number in a patch. This will be done at release time.

File: setup.py [hunk 2]
```diff
@@ -31,30 +31,31 @@
...
     'packages': ['rocker'],
     'package_dir': {'': 'src'},
     'package_data': {'rocker': ['templates/*.em']},
     'entry_points': {
         'console_scripts': [
             'rocker = rocker.cli:main',
             'detect_docker_image_os = rocker.cli:detect_image_os',
-	    ],
+        ],
```
Comment (tfoote) [inline setup.py:42]: please avoid unrelated changes to the PR.

File: src/rocker/extensions.py [hunk 3]
```diff
@@ -273,3 +273,26 @@ def register_arguments(parser, defaults={}):
...
+        return 'privileged'
+
+    def __init__(self):
+        self.name = Privileged.get_name()
+
+    def get_snippet(self, cli_args):
+        return ''
+
+    def get_docker_args(self, cli_args):
+        return '--privileged'
```
Comment (tfoote) [inline src/rocker/extensions.py:291]: This needs to be prepended with whitespace to make sure not to run into a preceding argument if not first. 
```suggestion
        return ' --privileged'
```

They are currently string appended https://github.com/osrf/rocker/blob/4f9f6d7595ccc23a2ae9b600a2262deefb676d89/src/rocker/core.py#L257 This could be made more robust with a ' '.join(...) it looks like other places have this potential problem depending on ordering.

File: setup.py (hunk 2 above)
Comment (tfoote) [reply setup.py:42]: A little cleanup can be done, but it should be at least separated to a different commit.

**Review by tfoote — CHANGES_REQUESTED** [review]
Thanks this will be useful to add. I had a few requests to clean up the patch but otherwise it looks good.

File: setup.py (hunk 1 above)
Comment (galou) [reply setup.py:34]: My point of view on this is that we add a functionality and thus must increment the minor version number but will obey to your request. I'll increment first RP #130.

File: src/rocker/extensions.py (hunk 3 above)
Comment (galou) [reply src/rocker/extensions.py]: Wouldn't it be better to actually let extensions return a list of strings and let core.py manage formatting the command, if needed?

File: setup.py (hunk 1 above)
Comment (tfoote) [reply setup.py]: Yes, that's the policy for semantic versioning. But this commit is not the release. There may be multiple additions that go into the next release. And I couldn't merge this PR at the same time as your other one because this change would conflict.

File: src/rocker/extensions.py (hunk 3 above)
Comment (tfoote) [reply src/rocker/extensions.py]: I think a single string at least with core.py inserting the space. "If needed" is not clearly resolved if I return multiple.

File: setup.py (hunk 1 above)
Comment (galou) [reply setup.py]: I get it with the version number. I actually never contributed to a release-based repository. How do you then manage the version change? Do contributors have to notify somewhere that they did a change that provokes a version change?

File: src/rocker/extensions.py (hunk 3 above)
Comment (galou) [reply src/rocker/extensions.py]: There might be a few words missing in these last sentences and I don't get their sense. Anyway, with "if needed", I meant that `subprocess.run` and partially `pexcpect.spawn` accept a list of strings as argument to launch the command.

**Comment by galou** [comment]
Reverted the version number and unrelated changes. Should I squeeze commits?

**Review by tfoote — APPROVED** [review]
Thanks for cleaning it up. I added some tests too. I'll just wait for CI to return. Don't worry about cleaning up the commit history, I'll squash when merging.

**Comment by tfoote** [comment]
Coverage isn't reporting directly: 100% diff coverage now https://app.codecov.io/gh/osrf/rocker/compare/132/overview
